src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -74,7 +74,21 @@ private BoundExpression ReplaceTerminalElementsWithTemps(BoundExpression expr, T
             // Examples:
             // in `expr == (..., ...)` we need to save `expr` because it's not a tuple literal
             // in `(..., expr) == (..., (..., ...))` we need to save `expr` because it is used in a simple comparison
-            return EvaluateSideEffectingArgumentToTemp(VisitExpression(expr), initEffects, ref temps);
+            BoundExpression loweredExpr = VisitExpression(expr);
+            if ((bool)(loweredExpr.Type != null) && NullableAlwaysHasValue((BoundExpression)loweredExpr) is var value && value != null)
+            {
+                // Optimization: if the nullable expression always has a value, we'll save the value to a temp and wrap
+                // that temp so it can be recognized as always having a value later on.
+                var savedValue = EvaluateSideEffectingArgumentToTemp(value, initEffects, ref temps);
+                SyntaxNode syntax = loweredExpr.Syntax;
+                _ = TryGetSpecialTypeMethod(syntax, SpecialMember.System_Nullable_T__ctor, out MethodSymbol nullableCtor);
```

Comment:
> _ = TryGetSpecialTypeMethod(syntax, SpecialMember.System_Nullable_T__ctor, out MethodSymbol nullableCtor); [](start = 15, length = 107)

I think, there is no need to go through all this trouble. Just update the argument in, loweredExpr, it is already an object creation, NullableAlwaysHasValue made sure of that. Especially that we are not supposed to ignore value returned by TryGetSpecialTypeMethod